client/package.json

```diff
@@ -205,12 +205,22 @@
           "autoDetect",
           "${cwd}"
         ],
-        "pattern": {
-          "regexp": "^.*ParseError (at|in) (.+?):(\\d+)?:? (.+)$",
-          "file": 2,
-          "line": 3,
-          "message": 4
-        }
+        "pattern": [
+          {
+            "regexp": "^.*ParseError (at|in) (.+?):(\\d+)?:? (unparsed line: '(.+)')$",
```

Comment:
I think the error can also arrive without the "unparsed line" extra info. We should keep maching it if that's the case.

Reply:
It's a pain to exhaustively generate all the kinds of errors. I had to look at the code, and try things randomly to get this list. I also lost the files I was using. Don't do the same mistake, version all files to produce these errors!

Reply:
Added files with one error code in each. 

And to clarify the process when we are going to use the BBMASK to ignore the error files:
1. Have files with one error code in each
2. Have BBMASK to ignore all the error files but the one we are testing
3. Overwrite the value in BBMASK with the next error file to test
4. Repeat 2 and 3 until all error files are tested

Reply:
Also there was a CI failure, I think the diagnostics are somewhat unstable, the number of diagnostics is not consistent between CIs since I added those error files

Reply:
If the diagnostics concern Python, you might get embedded languages diagnostics as well. You can check the "source" of the diagnostic and use assertWillComeTrue() to wait for the diagnostic you are expecting.

Reply:
Updated tests with BBMASK, but had some trouble looping them in a single test due to some potential async issues. Having one test for each file would work but as you said it takes a lot of time.

Reply:
Any update to BBMASK will trigger a full parsing which is slow anyway. How long does it get?

Reply:
Had another idea. Instead of BBMASK, we want to run `bitbake -b <error.bb> -p`. The difficulty now would be passing the `-b` argument to a parsing task..

Reply:
the test time varies, from as low as 10 sec to as high as 1 min according to the logs. but usually it takes ~ 45 seconds

Reply:
I'd just test one of the examples. If we really don't want to leave the rest untested, we could have unit tests for the regex. Although it kind of misses the point of testing the errors.

Reply:
Works for me.